### PR TITLE
♻️(frontend) improve networking operations

### PR DESCRIPTION
**Pull request by lebaudantoine**
## Purpose

Enhance the UX while loading the dashboard.

## Proposal

- [X] Fix refresh access token mechanism
- [X] Load comparison data last

https://github.com/openfun/warren/assets/45729124/0f53efb6-b892-496d-ab4f-b4cc529dba25

### CODE REVIEW

**Comment by lebaudantoine** [comment]
Some optimizations may be required. Currently, requests are executed in parallel, which leads to each request triggering a refresh token request. I need to investigate whether there are any design issues in the current implementation.

- [ ] Explore options to share the refresh token between parallel requests.
- [ ] Consider executing comparison requests in a sequential order.

**Review by jmaupetit — APPROVED** [review]

File: src/frontend/packages/ui/video/components/Metrics/ViewsMetric.tsx [hunk 1]
```diff
@@ -66,13 +71,24 @@ export const ViewsMetric: React.FC<ViewsMetricProps> = ({
...
     [since, until],
   );
-  const { videoViews: comparisonData, isFetching: isComparisonFetching } =
-    useVideosViews(videoIds, {
-      since: comparisonSince,
-      until: comparisonUntil,
-      ...params,
-    });
 
+  // Wait until the before last video views has resolved
```
[the remark this answers is not in the document]
Comment (lebaudantoine) [reply src/frontend/packages/ui/video/components/Metrics/ViewsMetric.tsx:75]: "avant dernier"

[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/frontend/packages/ui/video/components/Metrics/ViewsMetric.tsx:75]: yup :sweat_smile:

**Pull request merged**